tests/integration-service/integration.go

```diff
@@ -263,6 +265,66 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
 			Expect(f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot)).To(BeFalse(), "expected tests to fail for snapshot %s/%s", snapshot.GetNamespace(), snapshot.GetName())
 		})
 
+		It("creates a new IntegrationTestScenario", FlakeAttempts(3), func() {
+			newIntegrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario(applicationName, testNamespace, BundleURL, InPipelineName)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("updates the Snapshot with the re-run label for the new scenario", FlakeAttempts(3), func() {
+			updatedSnapshot := snapshot.DeepCopy()
+			err := metadata.AddLabels(updatedSnapshot, map[string]string{snapshotRerunLabel: newIntegrationTestScenario.Name})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(f.AsKubeAdmin.IntegrationController.PatchSnapshot(snapshot, updatedSnapshot)).Should(Succeed())
+			Expect(metadata.GetLabelsWithPrefix(updatedSnapshot, snapshotRerunLabel)).NotTo(BeEmpty())
+		})
+
+		When("An snapshot is updated with a rerun label for a given scenario", func() {
+			It("checks if the new integration pipelineRun started", Label("slow"), func() {
+				reRunPipelineRun, err := f.AsKubeDeveloper.IntegrationController.WaitForIntegrationPipelineToGetStarted(newIntegrationTestScenario.Name, snapshot.Name, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(reRunPipelineRun).ShouldNot(BeNil())
+			})
+
+			It("checks if the rerun label was removed from the Snapshot", FlakeAttempts(3), func() {
+				Eventually(func() error {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, "", "", testNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					Expect(metadata.GetLabelsWithPrefix(snapshot, snapshotRerunLabel)).To(BeEmpty())
```

Comment:
I think that when this assertion fails, it will also fail the test (instead of retrying).
To make the retries work in case of the presence of the rerun label, we should return a non-nil error ([like this](https://github.com/redhat-appstudio/e2e-tests/blob/main/tests/integration-service/integration.go#L73-L75))

The same goes for other assertions within the Eventually blocks.

So I think we should only do those assertions within the Eventually block, which we consider a failure and should fail the test immediately. Wdyt?

Reply:
Ack, updated this Eventually block to not have assertions. 
There is also a [special case for Eventually blocks](https://pkg.go.dev/github.com/onsi/gomega#Eventually) that returns no values which supports Expect assertions and will rerun the function until all assertions pass. I think that one makes sense for checking if the integration test is reported in the Snapshot.

Reply:
Thanks for letting me know, I didn't know about that special cases before.